### PR TITLE
Fix/paypal discounts

### PR DESCRIPTION
**Pull request by maciekjablonski**
Like: https://github.com/mirumee/saleor/pull/707

### CODE REVIEW

**Comment by maciekjablonski** [comment]
Refs #727

**Comment by patrys** [comment]
I have this tingling feeling that you're fixing this in the wrong repository. This is a workaround, not a solution.

**Comment by artursmet** [comment]
@patrys But django-payments doesn't deliver any API to mark if ordered item product is a voucher or not. So I think it's good place to fix it here, instead of doing major refactoring of django-payments.

**Comment by patrys** [comment]
It should likely automatically insert a discount row based on the totals. The question is how do we provide the name and value (so we can double-check that it sums up to the correct value).

**Comment by maciekjablonski** [comment]
@krzysztofwolski @patrys could you see it after rebase?

**Comment by patrys** [comment]
I still believe this should be fixed in `django-payments`.

**Comment by patrys** [comment]
I apologize but I have to back down on this one. I did some research and it seems the most straightforward way is for Saleor to always report discounts as an order item with a negative discount. None of the gateways seem to have a dedicated support for discounts that would be worth adding yet another API to django-payments.

**Comment by krzysztofwolski** [comment]
@patrys :+1: 
Lookin good:
paypaldiscount

File: saleor/order/models.py [hunk 1]
```diff
@@ -404,6 +404,17 @@ def get_purchased_items(self):
             price=item.unit_price_gross.quantize(Decimal('0.01')),
             currency=settings.DEFAULT_CURRENCY)
             for item in self.order.get_items()]
+
+        # PayPal require discount info as item with negative price.
+        provider, _ = settings.PAYMENT_VARIANTS.get(self.variant, (None, None))
+        voucher = self.order.voucher
+        if provider and 'PaypalProvider' in provider and voucher is not None:
```
Comment (patrys) [inline saleor/order/models.py]: This should not be PayPal-specific.

**Pull request merged**